openpype/hosts/nuke/api/lib.py

```diff
@@ -2091,6 +2092,31 @@ def set_root_colorspace(self, imageio_host):
         workfile_settings.pop("colorManagement", None)
         workfile_settings.pop("OCIO_config", None)
 
+        # treat monitor lut separately
+        monitor_lut = workfile_settings.pop("monitorLut", None)
+        m_display, m_viewer = get_viewer_config_from_string(monitor_lut)
+        v_display, v_viewer = get_viewer_config_from_string(
+            viewer_process_settings
+        )
+
+        # set monitor lut differently for nuke version 14
+        if nuke.NUKE_VERSION_MAJOR >= 14:
+            workfile_settings["monitorOutLUT"] = create_viewer_profile_string(
+                m_viewer, m_display, path_like=False)
+            # monitorLut=thumbnails - viewerProcess makes more sense
+            workfile_settings["monitorLut"] = create_viewer_profile_string(
+                v_viewer, v_display, path_like=False)
+
+        if nuke.NUKE_VERSION_MAJOR == 13:
+            workfile_settings["monitorOutLUT"] = create_viewer_profile_string(
+                m_viewer, m_display, path_like=False)
+            # monitorLut=thumbnails - viewerProcess makes more sense
+            workfile_settings["monitorLut"] = create_viewer_profile_string(
+                v_viewer, v_display, path_like=True)
+        if nuke.NUKE_VERSION_MAJOR <= 12:
+            workfile_settings["monitorLut"] = create_viewer_profile_string(
+                m_viewer, m_display, path_like=True)
```

Comment:
Would it make more sense to allow the version specific details up to do the function that is separate?

```
get_viewer_profile_settings(m_viewer, m_display)
```

Then the function returns:
```python
return {
    "monitorOutLUT": a,
    "monitorLut": b
}
```
etc. based on the nuke version itself.
It feels slightly odd being embedded directly in the `set_root_colorspace` function which is already quite a lengthy function.

Reply:
good point, look here 33e18661241a59a612a06558fbc19f5de264220b